### PR TITLE
Add stubs for `python-http-client`

### PR DESCRIPTION
**Pull request by ajimenez2712**
Stubs for https://github.com/sendgrid/python-http-client.

### CODE REVIEW

File: stubs/python-http-client/python_http_client/__init__.pyi [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    NotFoundError as NotFoundError,
+    PayloadTooLargeError as PayloadTooLargeError,
+    ServiceUnavailableError as ServiceUnavailableError,
+    TooManyRequestsError as TooManyRequestsError,
+    UnauthorizedError as UnauthorizedError,
+    UnsupportedMediaTypeError as UnsupportedMediaTypeError,
+)
+
+dir_path: str
+__version__: str
```
Comment (srittau) [inline stubs/python-http-client/python_http_client/__init__.pyi:18]:
```suggestion
dir_path: Final[str]
__version__: Final[str]
```

(Needs import from `typing`.)

File: stubs/python-http-client/python_http_client/client.pyi [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    def status_code(self) -> int: ...
+    @property
+    def body(self): ...
+    @property
+    def headers(self) -> Message: ...
+    @property
+    def to_dict(self): ...
+
+class Client:
+    methods: set[str]
```
Comment (srittau) [inline stubs/python-http-client/python_http_client/client.pyi:15]:
```suggestion
    methods: Final[set[str]]
```

File: stubs/python-http-client/python_http_client/exceptions.pyi [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+class NotFoundError(HTTPError): ...
+class MethodNotAllowedError(HTTPError): ...
+class PayloadTooLargeError(HTTPError): ...
+class UnsupportedMediaTypeError(HTTPError): ...
+class TooManyRequestsError(HTTPError): ...
+class InternalServerError(HTTPError): ...
+class ServiceUnavailableError(HTTPError): ...
+class GatewayTimeoutError(HTTPError): ...
+
+err_dict: dict[int, type[HTTPError]]
```
Comment (srittau) [inline stubs/python-http-client/python_http_client/exceptions.pyi:25]:
```suggestion
err_dict: Final[dict[int, type[HTTPError]]]
```

**Review by srittau — CHANGES_REQUESTED** [review]
Thanks. I suggest to add a few `Final`s (see below.)

**Comment by ajimenez2712** [comment]
> Thanks. I suggest to add a few `Final`s (see below.)

Thanks for the review, I already made all suggested changes.

**Review by srittau — APPROVED** [review]

**Pull request merged**